This patch make sure that the compiled asm code get assembled using GNU stack,
which is not-executable, to allow flac to work with PAX-enabled systems.

The patch was submitted by Kevin F. Quinn in Gentoo bug #114984.

More information about GNU stack marking can be found in the documentation at
http://www.gentoo.org/proj/en/hardened/gnu-stack.xml .
Index: flac-1.1.2/src/libFLAC/ia32/cpu_asm.nasm
===================================================================
--- flac-1.1.2.orig/src/libFLAC/ia32/cpu_asm.nasm
+++ flac-1.1.2/src/libFLAC/ia32/cpu_asm.nasm
@@ -98,3 +98,7 @@ cident FLAC__cpu_info_sse_test_asm_ia32
 	pop	ebx
 
 end
+
+%ifdef OBJ_FORMAT_elf
+       section .note.GNU-stack noalloc
+%endif
Index: flac-1.1.2/src/libFLAC/ia32/fixed_asm.nasm
===================================================================
--- flac-1.1.2.orig/src/libFLAC/ia32/fixed_asm.nasm
+++ flac-1.1.2/src/libFLAC/ia32/fixed_asm.nasm
@@ -304,3 +304,7 @@ cident FLAC__fixed_compute_best_predicto
 	ret
 
 end
+
+%ifdef OBJ_FORMAT_elf
+       section .note.GNU-stack noalloc
+%endif
Index: flac-1.1.2/src/libFLAC/ia32/lpc_asm.nasm
===================================================================
--- flac-1.1.2.orig/src/libFLAC/ia32/lpc_asm.nasm
+++ flac-1.1.2/src/libFLAC/ia32/lpc_asm.nasm
@@ -1501,3 +1501,7 @@ cident FLAC__lpc_restore_signal_asm_ia32
 	ret
 
 end
+
+%ifdef OBJ_FORMAT_elf
+       section .note.GNU-stack noalloc
+%endif
